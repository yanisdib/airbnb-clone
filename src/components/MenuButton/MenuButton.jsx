import PropTypes from 'prop-types';
import styled from 'styled-components';

import { AccountCircle, Menu } from '@styled-icons/material';


const Wrapper = styled.div`
    display: flex !important;
    justify-content: center !important;
    align-items: center !important;
`;

const Button = styled.button`
    display: flex !important;
    column-gap: 8px !important;
    align-items: center !important;
    justify-content: center !important;
    height: fit-content !important;
    background-color: #FFFFFF !important;
    border: 1px #DDDDDD solid !important;
    border-radius: 5rem !important;
    padding: 3px 3px 3px 10px !important;
    cursor: pointer !important;
`;

const UserPictureThumbnail = styled.div`
    display: flex !important;
`;

const UserIcon = styled(AccountCircle)`
    height: 34px !important;
    fill: #717171 !important;
`;

const MenuIcon = styled(Menu)`
    height: 19px !important;
    fill: #222222 !important;
`;


function MenuButton({ onClick }) {
    return (
        <Wrapper>
            <Button onClick={onClick}>
                <MenuIcon />
                <UserPictureThumbnail>
                    <UserIcon />
                </UserPictureThumbnail>
            </Button>
        </Wrapper>
    );
}


MenuButton.defaultProps = { onClick: () => { } };

MenuButton.propTypes = { onClick: PropTypes.func.isRequired };


export default MenuButton;