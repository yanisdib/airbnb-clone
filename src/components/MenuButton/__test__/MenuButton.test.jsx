import { shallow } from 'enzyme';

import MenuButton from '../MenuButton';


describe('MenuButton component', () => {
    let wrapper;

    beforeEach(() => wrapper = shallow(<MenuButton />))

    it('renders correctly without crashing', () => expect(wrapper).toMatchSnapshot())
})