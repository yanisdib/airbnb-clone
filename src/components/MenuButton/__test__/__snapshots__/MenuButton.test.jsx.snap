// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton component renders correctly without crashing 1`] = `ShallowWrapper {}`;
